propsigner/streamproto.go

```diff
@@ -0,0 +1,55 @@
+package propsigner
```

Comment:
Here're are the mentioned `readMsg` and `writeMsg`.

The good part of doing this simple scheme is that is very easy to port this remote signing protocol to other languages (JS, etc); also very efficient and easy to debug. (compared maybe to try doing gRPC on top of libp2p streams).

Reply:
I like the simplicity of your approach, but throwing another idea that would also be good for multi-language support: use https://github.com/libp2p/go-libp2p-gostream with HTTP (not gRPC like in go-threads).